File: examples-sdk/python/020-basic-ai-ab-testing.py
#!/usr/bin/env python3
"""020-basic-ai-ab-testing — Python SDK equivalent
Compile: vil compile --from python --input 020-basic-ai-ab-testing.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("ai-ab-testing-gateway", port=8080)
ab = server.service_process("ab")
ab.endpoint("POST", "/infer", "infer")
ab.endpoint("GET", "/metrics", "metrics")
ab.endpoint("POST", "/config", "update_config")
root = server.service_process("root")
root.endpoint("GET", "/", "index")
server.compile()

File: examples-sdk/typescript/020-basic-ai-ab-testing.ts
#!/usr/bin/env tsx
// 020-basic-ai-ab-testing — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 020-basic-ai-ab-testing.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("ai-ab-testing-gateway", 8080);
const ab = new ServiceProcess("ab");
ab.endpoint("POST", "/infer", "infer");
ab.endpoint("GET", "/metrics", "metrics");
ab.endpoint("POST", "/config", "update_config");
server.service(ab);
const root = new ServiceProcess("root");
root.endpoint("GET", "/", "index");
server.service(root);
server.compile();
